k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java

```diff
@@ -1002,11 +1021,23 @@ public void onBackPressed() {
             if (draftId == INVALID_DRAFT_ID) {
                 onDiscard();
             } else {
-                super.onBackPressed();
+                if (navigateUp) {
+                    openInboxFolder();
+                } else {
+                    super.onBackPressed();
+                }
             }
         }
     }
 
+    private void openInboxFolder() {
+        String folder = account.getAutoExpandFolderName();
+        LocalSearch search = new LocalSearch(folder);
+        search.addAccountUuid(account.getUuid());
+        search.addAllowedFolder(folder);
+        MessageList.actionDisplaySearch(this, search, false, false);
```

Comment:
This activity can be launched by third-party apps into their own task. This means you need to use `true` for the last argument so `MessageList` is launched into K-9 Mail's task. And then you'll need to `finish()` this activity.